### PR TITLE
Add config object with validation and docstrings

### LINKED ISSUE #157: Config object
It could be interesting to make a dedicated `Config` class. This would provide the following advantages:

* It can do some checks whether the configuration from the yaml file is valid.
* It can be documented with inline comments in the code. This makes it easy to keep the docs in sync with the code.
* It makes it possible to set default options (though we might want to force the user to think hard about all settings)
* It would make the example config file less important, we could consider removing it from the repo altogether. This avoids syncing issues between the code and the example config.

A config class is probably most easily implemented as a dataclass, something like so:

```py
from dataclasses import dataclass

@dataclass
class Config:
    a: str  # required parameter, needs to be a string
    """Every parameter can be documented with inline comments.

    These comments can be used to build automatic documentation.
    """

    b: int = 5
    """Optional parameter with a default value of 5."""

    c: bool = False
    """The type will be checked when the data is first loaded."""

with open(config_file) as f:
    config_file = yaml.safe_load(f)

config = Config(**config_file)
```

### PR DESCRIPTION
**Pull request by Peter9192**
This PR adds a `Config` class to the model code. It is used as such:

```python
from wam2layers.config import Config

# This loads the settings and also does some validation, 
# e.g. whether a paths exist or a date is correctly formatted.
config = Config.from_yaml('config-file.yaml')

# Settings can now be accessed with dot notation instead of brackets:
print(config.target_frequency)
```

In addition, there is now extensive documentation on the configuration options. See the [preview](https://wam2layers--165.org.readthedocs.build/en/165/config.html)

This documentation is generated automatically based on the docstrings in `config.py` (using [sphinx-autodoc](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html))

The PR also removes the "/backtrack" in the path of the output folder (see #137), and removes the example config files.
Dates in existing config files should be reformatted (YYYY-MM-DD instead of YYYYMMDD).

closes #157

### CODE REVIEW

**Comment by Imme1992** [comment]
Great, thanks Peter, the explanation of the config file looks very good.
I have a few minor comments on the documentation.

Shall we put the target_frequency to 10min. I believe that is a better internal timestep to start with.
Periodic_boundary: Add there that this should be true when working with global data.

**Review by Imme1992 — APPROVED** [review]

**Review by ruudvdent — CHANGES_REQUESTED** [review]
Looks great, but I would be in favor to keep an updated example config file somewhere.

**Comment by Peter9192** [comment]
Thanks for the feedback Ruud! I will keep one `example-config.yaml` in the top folder. 

Looking again at it, I wonder whether we should

- Change the order of the options
- Remove options that are currently not available

File: docs/userguide.md [hunk 1]
```diff
@@ -237,8 +237,8 @@ wam2layers backtrack config-file.yaml
 where `config-file.yaml` is the path to your configuration file. Among others,
 this file should have settings on the date range for which you want to run
 track, and also about the location where the preprocessed data are stored and
-where the output will be stored. [Documentation on this](./config) is underway,
-but for now you can have a look at the example config file
+where the output will be stored. For more information, see [](./config) or have
+a look at the example config file
 [here](https://github.com/WAM2layers/WAM2layers/blob/main/cases/era5_2021.yaml).
```
[the remark this answers is not in the document]
Comment (Peter9192) [reply docs/userguide.md]: Well-spotted! Now updated with a new example-config file

[the remark this answers is not in the document]
Comment (Peter9192) [reply docs/userguide.md]: New link should work once this PR is merged

**Review by ruudvdent — APPROVED** [review]

**Pull request merged**